### PR TITLE
Add api, update validate flow create scaler

### PR DESCRIPTION
**Pull request by vuducmanh11**

### CODE REVIEW

**Comment by vtdat** [comment]
First thing first, could you make the title more clear?

File: api/scaler.go [hunk 1]
```diff
@@ -184,6 +184,87 @@ func (a *API) deleteScaler(w http.ResponseWriter, req *http.Request) {
...
+		return
+	}
+	scaler = make(map[string]model.Scaler, len(resp.Kvs))
+	for _, ev := range resp.Kvs {
+		var s model.Scaler
+		_ = json.Unmarshal(ev.Value, &s)
+		s.Active = true
+		scaler[string(ev.Key)] = s
+		js, err := json.Marshal(&s)
+		fmt.Println(js)
```
Comment (vtdat) [inline api/scaler.go]: Don't print pls.

File: api/scaler.go [hunk 2]
```diff
@@ -184,6 +184,87 @@ func (a *API) deleteScaler(w http.ResponseWriter, req *http.Request) {
...
+		})
+		return
+	}
+	scaler = make(map[string]model.Scaler, len(resp.Kvs))
+	for _, ev := range resp.Kvs {
+		var s model.Scaler
+		_ = json.Unmarshal(ev.Value, &s)
+		s.Active = true
+		scaler[string(ev.Key)] = s
+		js, err := json.Marshal(&s)
```
Comment (vtdat) [inline api/scaler.go]: should be `raw` or smt, `js` usually stands for javascript.

File: api/scaler.go [hunk 3]
```diff
@@ -184,6 +184,87 @@ func (a *API) deleteScaler(w http.ResponseWriter, req *http.Request) {
...
+			err:  err,
+		})
+		return
+	}
+	scaler = make(map[string]model.Scaler, len(resp.Kvs))
+	for _, ev := range resp.Kvs {
+		var s model.Scaler
+		_ = json.Unmarshal(ev.Value, &s)
+		s.Active = true
+		scaler[string(ev.Key)] = s
```
Comment (vtdat) [inline api/scaler.go]: you're not using `scaler`? then remove it.

File: api/scaler.go [hunk 4]
```diff
@@ -184,6 +184,87 @@ func (a *API) deleteScaler(w http.ResponseWriter, req *http.Request) {
 	return
 }
 
+// Active scaler
+func (a *API) activeScaler(w http.ResponseWriter, req *http.Request) {
```
Comment (vtdat) [inline api/scaler.go]: function name should be `activateScaler`

File: pkg/model/actionhttp.go [hunk 5]
```diff
@@ -44,5 +36,13 @@ func (a *ActionHTTP) Validate() error {
...
 	}
 
+	if err := a.validate(); err != nil {
+		return err
+	}
+
+	if err := a.URL.Validate(); err != nil {
+		return err
+	}
+
```
Comment (vtdat) [inline pkg/model/actionhttp.go]: why?

File: api/scaler.go [hunk 6]
```diff
@@ -184,6 +184,87 @@ func (a *API) deleteScaler(w http.ResponseWriter, req *http.Request) {
...
+		_ = json.Unmarshal(ev.Value, &s)
+		s.Active = false
+		scaler[string(ev.Key)] = s
+		js, _ := json.Marshal(&s)
+		_, err = a.etcdcli.DoPut(path, string(js))
+	}
+
+	a.respondSuccess(w, http.StatusOK, nil)
+	return
+}
```
Comment (vtdat) [inline api/scaler.go]: same as above

File: pkg/model/actionhttp.go (hunk 5 above)
Comment (vuducmanh11) [reply pkg/model/actionhttp.go]: if you validate before setting default value for a.Delay and a.DelayType, you will get errors:   "time.parseduration invalid", "unsupported delay type"

**Comment by vtdat** [comment]
Resolve without commits? @vuducmanh11

**Comment by vuducmanh11** [comment]
> Resolve without commits? @vuducmanh11

sr, i close this request and create new request

**Comment by ntk148v** [comment]
> > Resolve without commits? @vuducmanh11
> 
> sr, i close this request and create new request

image

hmmm :L

**Pull request closed without merging**